server/src/server.ts

```diff
@@ -399,6 +403,15 @@ function hover(msg: p.RequestMessage) {
 }
 
 function inlayHint(msg: p.RequestMessage) {
+
+  if (!initializationOptions.inlayHints?.enable) {
+    return {
+      jsonrpc: c.jsonrpcVersion,
+      id: msg.id,
+      result: null,
+    };
+  }
```

Comment:
To clear inlay hints when setting is disabled.

Reply:
Doing this, and not restarting the server, means the client will continue to send messages to the server for these features. The point of turning the features _off_ via a setting is also that they should be zero cost and not potentially slow down the extension. Even if this is just a null response, it'll still keep sending all of the messages back and forth. Might not hurt performance for inlay hints, but feature X, Y and Z that we'll add in the future might. Better to just turn it off completely via capabilities and a restart, like we're doing now. Then it's truly zero cost.